### DISCUSSION TITLE
Using [deepgram-python-sdk : deepgram.transcription.prerecorded ] getting error :  DG: 408, message='Request Timeout' frequently even for 7-8 min audio file as well.

### DISCUSSION THREAD

**Discussion by rohitashPrajapati** [deepgram/community#322] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Hi there,

First of all, thanks for creating such a great API for diarization and transcription. I was trying to use the deepgram-python-sdk for prerecorded files transcriptions using deepgram.transcription.prerecorded function. But not sure why I am frequently getting the below error:

`DG: 408, message='Request Timeout', url=URL('https://api.deepgram.com/v1/listen?tier=nova&model=general&language=en&smart_format=true&diarize=true&utterances=true&utt_split=0.5&punctuate=true')
Future exception was never retrieved
future: <Future finished exception=ClientOSError(32, 'Broken pipe')>
aiohttp.client_exceptions.ClientOSError: [Errno 32] Broken pipe`

I tried the code for audio with length 8 min it give error sometimes, but as the length and size of file increase it give error more frequently. I am not able to transcript the audio of length 0.5 hr or more.

I also tried to apply "timeout" greater than 300 in deepgram.transcription.prerecorded function, but still having the same issue, this timeout I am getting randomly sometimes more quickly and sometimes after 3-4 minutes.

I would greatly appreciate any help regarding this. thank you.

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

047a824d-da10-4ccc-a582-4ac8067aa9df

### If possible, please attach your code or paste it into the text box.

```

async def diarization_using_deepgram(upload_file: UploadFile):
    # Set the temp folder for all operations 
    tempFolderPath = getTempFolderPath()
    # Check for audio file
    if upload_file is None:
        return {'error': 'Audio file not found.'}
    # Save the uploaded file to disk
    filename = await save_audio_file_as_wave(upload_file, outputFolder=tempFolderPath)
    print(f"File saved successfully here: {datetime.now().strftime('%H:%M:%S')}")
    # Initialize the Deepgram SDK
    deepgram = Deepgram(Credentials.keys['deepgram_api_key'])
    
    retry_after_seconds = 5
    max_retries = 3
    retries = 0

    while retries < max_retries:
        try: 
            # file is local, Open the audio file
            audio = open(filename, 'rb')
            # Set the source
            source = {
                'buffer': audio,
                'mimetype': 'audio/wav'
            }

            #   # Send the audio to Deepgram and get the response
            data = await asyncio.create_task(
                deepgram.transcription.prerecorded(
                source,
                {
                    'model': "whisper-medium", # 'whisper-large',
                     # 'tier' : 'base',
                    # 'model' : 'meeting',
                    'language': 'en',
                    'smart_format': True,
                    'diarize': True,
                    'utterances': True,
                    'utt_split': 0.5,
                    # 'punctuate': True,
                }
                )
            )
            # print(f"data: ", data)
            # Try to remove the folder we created temporary
            removeFolder(tempFolderPath)  
            
            return data
        
        except IOError as e:
            if e.errno == errno.EPIPE:
                print(f"encountered an error: {e}")
                pass
        except Exception as e:
            print(f"Main function encountered an error: {e}", e)
            retries += 1
            time.sleep(retry_after_seconds)
            pass
```

**Comment by craigargh** [comment, marked as answer]
I've been having this issue as well. 

I believe it is related to this issue https://github.com/orgs/deepgram/discussions/272

The size of file that the API seems to be able to handle seems to vary between 10 mins and 10 seconds. My solution was to break the audio files into smaller segments and change the audio from stereo to mono. 

For most of the week I was processing 30 minute files in 10 second segments.
